File: frontend/src/auth/roles.js
// Canonical role vocabulary shared by authentication, navigation and role UI.
export const ROLES = Object.freeze({
    ADMIN: "Admin",
    SALES_EXECUTIVE: "Sales Executive",
    SALES_MANAGER: "Sales Manager",
    PRE_SALES_MANAGER: "Pre-Sales Manager",
    SOLUTION_ENGINEER: "Solution Engineer",
    DELIVERY: "Delivery",
});

export const AVAILABLE_ROLES = Object.freeze([
    ROLES.ADMIN,
    ROLES.SALES_EXECUTIVE,
    ROLES.SALES_MANAGER,
    ROLES.PRE_SALES_MANAGER,
    ROLES.SOLUTION_ENGINEER,
    ROLES.DELIVERY,
]);

export const isValidRole = (role) =>
    AVAILABLE_ROLES.includes(role);

File: app/constants/roles.py
# Canonical business roles for the Deal Room application.
#
# Keep these values stable across backend JWT claims, API payloads, database
# user-role rows, and frontend role checks.

ADMIN = "Admin"
SALES_EXECUTIVE = "Sales Executive"
SALES_MANAGER = "Sales Manager"
PRE_SALES_MANAGER = "Pre-Sales Manager"
SOLUTION_ENGINEER = "Solution Engineer"
DELIVERY = "Delivery"

AVAILABLE_ROLES = [
    ADMIN,
    SALES_EXECUTIVE,
    SALES_MANAGER,
    PRE_SALES_MANAGER,
    SOLUTION_ENGINEER,
    DELIVERY,
]

DEFAULT_ROLE = SALES_EXECUTIVE


def is_valid_role(role):
    """Return True when *role* is one of the six canonical roles."""
    return role in AVAILABLE_ROLES
